clusterctl/cmd/create_cluster.go

```diff
@@ -78,30 +79,35 @@ func RunCreate(co *CreateOptions) error {
 	if err != nil {
 		return err
 	}
+	ac, err := ioutil.ReadFile(co.AddonComponents)
+	if err != nil {
+		return err
+	}
 	pcsFactory := clusterdeployer.NewProviderComponentsStoreFactory()
 	d := clusterdeployer.New(
 		mini,
 		clusterdeployer.NewClientFactory(),
 		pd,
 		string(pc),
+		string(ac),
 		co.KubeconfigOutput,
 		co.CleanupExternalCluster)
-	err = d.Create(c, m, pcsFactory)
-	return err
+	return d.Create(c, m, pcsFactory)
 }
 
 func init() {
 	// Required flags
 	createClusterCmd.Flags().StringVarP(&co.Cluster, "cluster", "c", "", "A yaml file containing cluster object definition")
 	createClusterCmd.Flags().StringVarP(&co.Machine, "machines", "m", "", "A yaml file containing machine object definition(s)")
 	createClusterCmd.Flags().StringVarP(&co.ProviderComponents, "provider-components", "p", "", "A yaml file containing cluster api provider controllers and supporting objects")
+	createClusterCmd.Flags().StringVarP(&co.AddonComponents, "addon-components", "a", "", "A yaml file containing cluster addons to apply to the internal cluster")
 	// TODO: Remove as soon as code allows https://github.com/kubernetes-sigs/cluster-api/issues/157
 	createClusterCmd.Flags().StringVarP(&co.Provider, "provider", "", "", "Which provider deployment logic to use (google/vsphere)")
 
 	// Optional flags
 	createClusterCmd.Flags().BoolVarP(&co.CleanupExternalCluster, "cleanup-external-cluster", "", true, "Whether to cleanup the external cluster after bootstrap")
 	createClusterCmd.Flags().StringVarP(&co.VmDriver, "vm-driver", "", "", "Which vm driver to use for minikube")
-	createClusterCmd.Flags().StringVarP(&co.KubeconfigOutput, "kubeconfig-out", "", "kubeconfig", "where to output the kubeconfig for the provisioned cluster.")
+	createClusterCmd.Flags().StringVarP(&co.KubeconfigOutput, "kubeconfig-out", "", "kubeconfig", "where to output the kubeconfig for the provisioned cluster")
```

Comment:
Good that we made this consistent -- I'm of the inclination that we should have a period at the end of each of these descriptions so that the inevitable description that is more than one sentence won't look out of place. That can come later though.

Reply:
I just removed this one since it was out of place. Making them consistent with periods is fine with me though if you think that will cause less churn later. I'll fix it now since I have to regenerate all of the *.golden files as it is...

Reply:
I don't really like the way that they are output with periods at the end (since the default value is appended by the library code after our text) so I'm not going to do it now. We can debate that in a separate PR. :)

Reply:
Makes sense, wasn't aware of that.